src/Autofac/Features/OpenGenerics/OpenGenericDecoratorRegistrationSource.cs

```diff
@@ -77,13 +77,19 @@ public IEnumerable<IComponentRegistration> RegistrationsFor(Service service, Fun
             return Enumerable.Empty<IComponentRegistration>();
         }
 
-        private static IEnumerable<Parameter> AddDecoratedComponentParameter(Type decoratedParameterType, IComponentRegistration decoratedComponent, IEnumerable<Parameter> configuredParameters)
+        private static IList<Parameter> AddDecoratedComponentParameter(Type decoratedParameterType, IComponentRegistration decoratedComponent, IList<Parameter> configuredParameters)
         {
             var parameter = new ResolvedParameter(
                 (pi, c) => pi.ParameterType == decoratedParameterType,
                 (pi, c) => c.ResolveComponent(decoratedComponent, Enumerable.Empty<Parameter>()));
 
-            return new[] { parameter }.Concat(configuredParameters);
+            var resultArray = new Parameter[configuredParameters.Count + 1];
```

Comment:
No problem

Reply:
I brought all the changes from this file into the `develop` branch.